cloudferrylib/os/compute/usage_quota.py

```diff
@@ -0,0 +1,126 @@
+# Copyright (c) 2014 Mirantis Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the License);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an AS IS BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and#
+# limitations under the License.
+from cloudferrylib.utils import log
+
+
+LOG = log.getLogger(__name__)
+
+
+class UsageQuotaCompute(object):
+    """
+     - Change user_id
+     - Decrement usage quotas:
+        - instances
+        - vcpus
+        - RAM
+        - Sec Grs
+     - Increment usage quotas
+        - instances
+        - vcpus
+        - RAM
+        - Sec Grs
+    """
+
+    NOVA_INSTANCES = "nova.instances"
+    NOVA_QUOTA_USAGES = "nova.quota_usages"
+    NOVA_SEC_GRS = "nova.security_groups"
+
+    @staticmethod
+    def change_usage_quota_instance(sql_connector, flavor,
+                                    old_user_id, user_id, tenant_id):
+        vcpus = flavor['vcpus']
+        ram = flavor['ram']
+        UsageQuotaCompute.dec_quota_usage(sql_connector, old_user_id,
+                                          tenant_id, {'instances': 1,
+                                                      'cores': vcpus,
+                                                      'ram': ram})
+        UsageQuotaCompute.inc_quota_usage(sql_connector, user_id,
+                                          tenant_id, {'instances': 1,
+                                                      'cores': vcpus,
+                                                      'ram': ram})
+
+    @classmethod
+    def change_user_owner_sec_grs(cls, db, security_group_id, new_user_id):
+        sql = "UPDATE {sec_groups} SET " \
+              "{sec_groups}.user_id = '{new_user_id}' " \
```

Comment:
Also, do not use \<table name>.\<field name> if you are not using several tables in one query.